### PR TITLE
code review suggestions 2

### PR DESCRIPTION
**Pull request by corinmanning2315**
Description/Analysis of the Project
This is an app that helps you find recycling centers and save your favorite centers.

Was the program available in UC Github on time?
Yes

Is the program documented/commented well enough for you to understand?
Yes

Does the program compile?
Yes

Rationale Behind Changes
I just changed a few things so that the project met best practices standards. There were no noticeable errors. I just made a couple of suggestions.

3 Commits
1. https://github.com/ahlersml/MyCleaningLog/commit/2cd782f
2. https://github.com/ahlersml/MyCleaningLog/commit/9cd8e3c
3. https://github.com/ahlersml/MyCleaningLog/commit/0bfb7cd

Things I Learned
1. The difference between a list and arraylist (and how they connect)
2. Why inline comments are not preferred
3. How to implement retrofit

### CODE REVIEW

File: app/src/main/java/com/findmyrecycling/MainActivity.kt [hunk 1]
```diff
@@ -19,7 +19,6 @@ class MainActivity : ComponentActivity() {
         super.onCreate(savedInstanceState)
         setContent {
             FindMyRecyclingTheme {
-                // A surface container using the 'background' color from the theme
```
Comment (corinmanning2315) [inline app/src/main/java/com/findmyrecycling/MainActivity.kt:22]: removed inline comment 
https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

File: app/src/main/java/com/findmyrecycling/ProfileScreen.kt [hunk 2]
```diff
@@ -18,9 +18,7 @@ class ProfileScreen : ComponentActivity() {
         super.onCreate(savedInstanceState)
         setContent {
             FindMyRecyclingTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(color = MaterialTheme.colors.background,
-                    modifier = Modifier.fillMaxWidth()) {
+                Surface(color = MaterialTheme.colors.background, modifier = Modifier.fillMaxWidth()) {
```
Comment (corinmanning2315) [inline app/src/main/java/com/findmyrecycling/ProfileScreen.kt:21]: condensed to one line

File: app/src/main/java/com/findmyrecycling/RetrofitClientInstance.kt [hunk 3]
```diff
@@ -9,9 +9,7 @@ object RetrofitClientInstance {
 
     val retrofitInstance : Retrofit?
         get() {
-            // has this object been created yet?
             if (retrofit == null) {
-                // create it
```
Comment (corinmanning2315) [inline app/src/main/java/com/findmyrecycling/RetrofitClientInstance.kt:14]: removed inline comment 
https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

File: app/src/main/java/com/findmyrecycling/RetrofitClientInstance.kt [hunk 4]
```diff
@@ -9,9 +9,7 @@ object RetrofitClientInstance {
 
     val retrofitInstance : Retrofit?
         get() {
-            // has this object been created yet?
```
Comment (corinmanning2315) [inline app/src/main/java/com/findmyrecycling/RetrofitClientInstance.kt:12]: removed inline comment 
https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 5]
```diff
@@ -23,7 +23,7 @@ class ProductTests {
     var rule: TestRule = InstantTaskExecutorRule()
 
     lateinit var productService : ProductService
-    var allProducts : List<Product>? = ArrayList<Product>()
+    private var allProducts : List<Product>? = ArrayList<Product>()
```
Comment (corinmanning2315) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:26]: made the variable private

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 6]
```diff
@@ -57,7 +57,7 @@ class ProductTests {
     }
 
     @Test
-    fun `Given product data are available when I search for giberish then I should receive nothing` () = runTest {
+    fun `Given product data are available when I search for gibberish then I should receive nothing` () = runTest {
```
Comment (corinmanning2315) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:60]: grammar

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 7]
```diff
@@ -88,7 +88,7 @@ class ProductTests {
         assertTrue(allProducts!!.isNotEmpty())
         var containsBatteries = false
         allProducts!!.forEach {
-            if (it.product == "Rechargable Batteries" || it.product.equals("Lead Acid Batteries")) {
+            if (it.product == "Rechargeable Batteries" || it.product.equals("Lead Acid Batteries")) {
```
Comment (corinmanning2315) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:91]: grammar

File: app/src/test/java/com/findmyrecycling/ProductTests.kt [hunk 8]
```diff
@@ -133,7 +133,6 @@ class ProductTests {
         var allProducts : List<Product>? = ArrayList<Product>()
 
         val latch = CountDownLatch(1)
-        //creating a var obersever assigning a static object
```
Comment (corinmanning2315) [inline app/src/test/java/com/findmyrecycling/ProductTests.kt:136]: removed inline comment 
https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

**Comment by discospiff** [comment]
From this review:
- Removing inline comments... I could go either way.  Code should be self-describing.  If not, that's where inline comments come in.
- Consolidating to one line: Some developers will do this; others do the opposite.  My recommendation: use the IDE to create formatting rules and apply them universally.  
-  == can be used in place of .equals() in Kotlin.  They will generally yield the same results.
Bottom line...  I encourage the group to review the recommendations here, though I don't know that there is enough tech debt reduction to justify merging this branch.
